de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/services/CommissionInstanceRequestFactory.java

```diff
@@ -114,4 +117,38 @@ private CreateInstanceRequest createRequest(
 				.build();
 		return request;
 	}
+
+	public Optional<CreateInstanceRequest> createRequestFor(final CreateForecastCommissionInstanceRequest retrieveForecastCommissionPointsRequest)
+	{
+		final Hierarchy hierarchy = commissionHierarchyFactory.createFor(retrieveForecastCommissionPointsRequest.getSalesRepId());
+
+		final ConfigRequestForNewInstance contractRequest = ConfigRequestForNewInstance.builder()
+				.customerBPartnerId( retrieveForecastCommissionPointsRequest.getCustomerId() )
+				.salesRepBPartnerId( retrieveForecastCommissionPointsRequest.getSalesRepId() )
+				.date( retrieveForecastCommissionPointsRequest.getDateOrdered() )
+				.salesProductId( retrieveForecastCommissionPointsRequest.getProductId() )
+				.commissionHierarchy(hierarchy)
+				.build();
+
+		final ImmutableList<CommissionConfig> configs =
+				commissionContractFactory.createForNewCommissionInstances(contractRequest)
+					.stream()
+				    .filter(config -> config.getContractFor( Beneficiary.of( contractRequest.getSalesRepBPartnerId() ) ) != null)
+				    .collect( ImmutableList.toImmutableList() );
+
+		Check.assume(configs.size() <= 1, "There is only one active commission contract for a sales rep at a certain time!");
```

Comment:
making it explicit like this is good, but please add more info, so the errormessage is as actionable as possilbe.
suggestion:
```suggestion
		if (configs.size() > 0)
		{
			throw new AdempiereException("Expecting only one active commissionConfig for a sales rep at a certain time!")
					.appendParametersToMessage()
					.setParameter("salesRepBPartnerId", contractRequest.getSalesRepBPartnerId())
					.setParameter("contractRequest", contractRequest)
					.setParameter("hierarchy", hierarchy)
					.setParameter("commissionConfigs", configs);
		}
```